code/game/sound.dm

```diff
@@ -288,118 +314,134 @@ var/list/rummage_sound = list(\
+				M.playsound_local(turf_source, soundin, vol, vary, frequency, falloff, is_global, extrarange, override_env, envdry, envwet)
 
 var/const/FALLOFF_SOUNDS = 0.5
 
-//turf_source = our_turf, soundin = 'sound/effects/alert.ogg', vol = 100, vary = 1, extrarange = 0.5
-/mob/proc/playsound_local(var/turf/turf_source, soundin, vol as num, vary, extrarange as num, frequency, falloff, is_global, use_pressure = TRUE)
-	if(!src.client || ear_deaf > 0)	return
-	soundin = get_sfx(soundin)
-
-	var/sound/S = sound(soundin)
-	S.wait = 0 //No queue
-	S.channel = 0 //Any channel
-	S.volume = vol
-	S.environment = -1
-	if (vary)
+/mob/proc/playsound_local(var/turf/turf_source, soundin, vol as num, vary, frequency, falloff, is_global, extrarange, override_env, envdry, envwet)
+	if(!src.client || ear_deaf > 0)
+		return
+
+	var/sound/S = soundin
+	if(!istype(S))
+		soundin = get_sfx(soundin)
+		S = sound(soundin)
+		S.wait = 0 //No queue
+		S.channel = 0 //Any channel
+		S.volume = vol
+		S.environment = -1
 		if(frequency)
 			S.frequency = frequency
-		else
+		else if (vary)
 			S.frequency = get_rand_frequency()
 
 	//sound volume falloff with pressure
 	var/pressure_factor = 1.0
-
-	// 3D sounds, the technology is here!
+	
 	var/turf/T = get_turf(src)
-
-	if(T)
+	// 3D sounds, the technology is here!
+	if(isturf(turf_source))
 		//sound volume falloff with distance
 		var/distance = get_dist(T, turf_source)
+
 		S.volume -= max(distance - (world.view + extrarange), 0) * 2 //multiplicative falloff to add on top of natural audio falloff.
 
 		var/datum/gas_mixture/hearer_env = T.return_air()
 		var/datum/gas_mixture/source_env = turf_source.return_air()
 
-		//Use pressure flag allows you to ignore the normal environment based checks, allowing sounds that can be heard in/from space
-		if (use_pressure)
-			if (hearer_env && source_env)
-				var/pressure = min(hearer_env.return_pressure(), source_env.return_pressure())
-				if (pressure < ONE_ATMOSPHERE)
-					pressure_factor = max((pressure - SOUND_MINIMUM_PRESSURE)/(ONE_ATMOSPHERE - SOUND_MINIMUM_PRESSURE), 0)
-			else //in space
-				pressure_factor = 0
+		if (hearer_env && source_env)
+			var/pressure = min(hearer_env.return_pressure(), source_env.return_pressure())
 
-			if (distance <= 1)
-				pressure_factor = max(pressure_factor, 0.15)	//hearing through contact
+			if (pressure < ONE_ATMOSPHERE)
+				pressure_factor = max((pressure - SOUND_MINIMUM_PRESSURE)/(ONE_ATMOSPHERE - SOUND_MINIMUM_PRESSURE), 0)
+		else //in space
+			pressure_factor = 0
 
-			S.volume *= pressure_factor
+		if (distance <= 1)
+			pressure_factor = max(pressure_factor, 0.15)	//hearing through contact
+
+		S.volume *= pressure_factor
+
+		if(istype(T,/turf/simulated) && istype(turf_source,/turf/simulated))
+			var/turf/simulated/sim_source = turf_source
+			var/turf/simulated/sim_destination = T
+			if(sim_destination.zone != sim_source.zone)
+				S.volume -= 30
```

Comment:
Oh, and this here: Nebula/bay has a lot of sound sources with high base volumes. This bit of code here will result in a lot of sounds being inaudible through walls if you're using TG standards for playsound volumes (this can be heard in the demo you posted, as well). This isn't a part of z-sound, but it might be worth replacing this with an envdry adjustment to avoid the headaches associated with bumping up the volumes of sound sources across the board to compensate for this feature.

Reply:
Duly noted.